### ISSUE TITLE
label prevents disabling Variables::ProhibitReusedNames

### ISSUE THREAD

**Issue by jwilk** [Perl-Critic/Perl-Critic#661]
```
$ cat reused-names 

use strict;
use warnings;

our $VERSION = '3.14';

my $a = 1;
LABEL:
while (my $a = <FOO>) {  ## no critic (ProhibitReusedNames)
}

$ perlcritic --verbose '%p at %l\n' -2 reused-names
Variables::ProhibitReusedNames at 10
Miscellanea::ProhibitUselessNoCritic at 10
```

So it reports both ProhibitReusedNames at line 10, and that there's no ProhibitReusedNames to disable on line 10...
If I remove the label, the problem goes away:

```
$ sed -e '/^LABEL/d' reused-names > reused-names-nolabel
$ perlcritic -2 reused-names-nolabel
reused-names-nolabel source OK
```

**Comment by mcreenan** [comment]
Is there a specific version of Perl-Critic you are using?  I am unable to reproduce this on the dev or master branches

**Comment by jwilk** [comment]
I'm using 1.126 on Debian unstable.
But I reproduced it also with git master (a72a4c72749157d2fbcbc5c387935d0af09c3225).

**Comment by mcreenan** [comment]
Can you post your .perlcriticrc?

**Comment by jwilk** [comment]
I don't have any.

**Comment by mcreenan** [comment]
I made an attempt to fix this, you can try with my version and let me know if it works for you.

**Comment by jwilk** [comment]
Yup, 81e32c329940d632965088259dd6a46ec6d5caee fixes the bug for me.
Thanks!

**Comment by carnil** [comment]
Any news on merging this in the development branch and a new Perl-Critic version?

**Comment by petdance** [comment]
@mcreenan It's good to see that you've got it working.  We need to get some tests added to the test suite that prove it works, and prove it doesn't break other cases as well, too.  Is that something you can pull together?

@carnil Are you being affected by this bug?  Can you post an example?

**Comment by carnil** [comment]
Hi

On Thu, Jul 06, 2017 at 06:50:17AM -0700, Andy Lester wrote:
> @carnil Are you being affected by this bug?  Can you post an
> example?

Not myself, but the case jwilk has reported is as well recorded
downstream in Debian at https://bugs.debian.org/734858 . The recently
released new upstream version triggered me to check our open
bugreports to see if any was fixed with the new version.

Regards,
Salvatore

**Comment by petdance** [comment]
I'm glad to look at it when someone can add tests to the test suite for it.
